### PR TITLE
[Doc] Improve the documentation for C++ style guide

### PR DESCRIPTION
**Pull request by k-ye**
Related issue = #

### CODE REVIEW

File: docs/lang/articles/contribution/cpp_style.md [hunk 1]
```diff
@@ -4,54 +4,31 @@ sidebar_position: 4
...
+- Provide default initializers to the class member variables, at least for the POD types.
+  ```cpp
+  class Foo {
+   private:
+    int x_{0};
+    char* buf_{nullptr};
+  };
+  ```
+- Embrace the smart pointers and avoid `new` and `delete`.
+- Avoid virtual function calls in the constructors or destructions ([explanation](https://wiki.sei.cmu.edu/confluence/display/cplusplus/OOP50-CPP.+Do+not+invoke+virtual+functions+from+constructors+or+destructors)).
```
Comment (k-ye) [inline docs/lang/articles/contribution/cpp_style.md:34]:
```suggestion
- Avoid virtual function calls in the constructors or destructors([explanation](https://wiki.sei.cmu.edu/confluence/display/cplusplus/OOP50-CPP.+Do+not+invoke+virtual+functions+from+constructors+or+destructors)).
```

**Comment by lin-hitonami** [comment]
https://github.com/taichi-dev/taichi/blob/7f3d4c33f2a7be9b7ce092d7f8efd8fac493a9d7/docs/lang/articles/contribution/writing_cpp_tests.md#L16-L22
Just running `./taichi_cpp_tests` will result in error like this: 
```
[E 09/26/21 11:00:59.747 298012] [lang_util.cpp:runtime_lib_dir@25] If you are running the taichi_cpp_tests please set TI_LIB_DIR to $TAICHI_INSTALL_DIR/lib where TAICHI_INSTALL_DIR can be retrieved from taichi.__path__ in python
```
Shall we add instructions to set `$TI_LIB_DIR` in this section?

File: docs/lang/articles/contribution/writing_cpp_tests.md [hunk 2]
```diff
@@ -18,12 +19,15 @@ cmake .. -DTI_BUILD_TESTS=ON # ... other regular Taichi cmake args
...
 
 # run the CPP test
-./taichi_cpp_tests
+TI_LIB_DIR=$TAICHI_INSTALL_DIR/lib ./taichi_cpp_tests
 ```
 
 :::note
-On Windows, `taichi_cpp_tests.exe` will be placed inside the `%TAICHI_REPO_DIR%\bin` directory,
-so you can directly run it after adding the directory to `PATH` in [Setting up Taichi for development](dev_install.md#setting-up-taichi-for-development-3).
+In order to run the CPP tests, please setup the environment variable, `$TI_LIB_DIR$`, to point to `$TAICHI_INSTALL_DIR/lib`. `$TAICHI_INSTALL_DIR` can be retrieved from `taichi.__path__` in python.
```
Comment (lin-hitonami) [inline docs/lang/articles/contribution/writing_cpp_tests.md:26]: Is `$TI_LIB_DIR$` a typo? Should it be `$TI_LIB_DIR` ?

Comment (k-ye) [reply docs/lang/articles/contribution/writing_cpp_tests.md]: nice catch!

File: docs/lang/articles/contribution/writing_cpp_tests.md [hunk 3]
```diff
@@ -18,12 +19,15 @@ cmake .. -DTI_BUILD_TESTS=ON # ... other regular Taichi cmake args
...
 
 # run the CPP test
-./taichi_cpp_tests
+TI_LIB_DIR=$TAICHI_INSTALL_DIR/lib ./taichi_cpp_tests
 ```
 
 :::note
-On Windows, `taichi_cpp_tests.exe` will be placed inside the `%TAICHI_REPO_DIR%\bin` directory,
-so you can directly run it after adding the directory to `PATH` in [Setting up Taichi for development](dev_install.md#setting-up-taichi-for-development-3).
+In order to run the CPP tests, please setup the environment variable, `$TI_LIB_DIR`, to point to `$TAICHI_INSTALL_DIR/lib`. `$TAICHI_INSTALL_DIR` can be retrieved from `taichi.__path__` in python.
```
Comment (ailzhang) [inline docs/lang/articles/contribution/writing_cpp_tests.md:26]: nit: `taichi.__path__[0]`

**Review by ailzhang — APPROVED** [review]
Awesome thanks!

File: docs/lang/articles/contribution/writing_cpp_tests.md (hunk 3 above)
Comment (k-ye) [reply docs/lang/articles/contribution/writing_cpp_tests.md]: Thanks!

File: docs/lang/articles/contribution/cpp_style.md [hunk 4]
```diff
@@ -4,54 +4,31 @@ sidebar_position: 4
...
+- Provide default initializers to the class member variables, at least for the POD types.
+  ```cpp
+  class Foo {
+   private:
+    int x_{0};
+    char* buf_{nullptr};
+  };
+  ```
+- Embrace the smart pointers and avoid `new` and `delete`.
+- Avoid virtual function calls in the constructors or destructors([explanation](https://wiki.sei.cmu.edu/confluence/display/cplusplus/OOP50-CPP.+Do+not+invoke+virtual+functions+from+constructors+or+destructors)).
```
Comment (k-ye) [inline docs/lang/articles/contribution/cpp_style.md:34]:
```suggestion
- Avoid virtual function calls in the constructors or destructors ([explanation](https://wiki.sei.cmu.edu/confluence/display/cplusplus/OOP50-CPP.+Do+not+invoke+virtual+functions+from+constructors+or+destructors)).
```

**Review by strongoier — APPROVED** [review]
LGTM! Just a little concern: is it common to use `CPP` in a formal doc to call the C++ language?

**Pull request merged**